cpp/src/io/text/multibyte_split.cu

```diff
@@ -708,21 +679,40 @@ std::unique_ptr<cudf::column> multibyte_split(cudf::io::text::data_chunk_source
         tile_offsets,
         {device_delim.data(), static_cast<std::size_t>(device_delim.size())},
         *chunk,
-        byte_range_end,
         offset_output);
     }
 
     // load the next chunk
     auto next_chunk = reader->get_next_chunk(ITEMS_PER_CHUNK, read_stream);
     // while that is running, determine how many offsets we output (synchronizes)
-    auto next_tile_offset =
-      tile_offsets.get_inclusive_prefix(base_tile_idx + tiles_in_launch - 1, scan_stream);
-    offset_storage.advance_output(next_tile_offset.offset() - offset_storage.size());
+    auto new_offsets_unclamped =
+      tile_offsets.get_inclusive_prefix(base_tile_idx + tiles_in_launch - 1, scan_stream) -
+      static_cast<int64_t>(offset_storage.size());
+    auto new_offsets = [&] {
+      // if we are not in the last chunk, we can use all offsets
+      if (chunk_offset + static_cast<int64_t>(chunk->size()) < byte_range_end) {
+        return new_offsets_unclamped;
```

Comment:
If `new_offsets_unclamped` is only used in the lambda, why do we define it outside of this lambda? I can see the argument for naming it, but it seems out of place above the lambda.

Reply:
I agree that it belongs into the lambda, I didn't notice that I no longer use it outside.